internal/security/fileprovider/provider_test.go

```diff
@@ -247,51 +247,12 @@ func TestNoDefaultsCustomTokenParameters(t *testing.T) {
 
 // TestTokenUsingDefaults
 func TestTokenUsingDefaults(t *testing.T) {
-	// Arrange
-	mockLogger := logger.MockLogger{}
-
-	mockFileIoPerformer := &MockFileIoPerformer{}
-	expectedService1Dir := filepath.Join(outputDir, "myservice")
-	expectedService1File := filepath.Join(expectedService1Dir, outputFilename)
-	service1Buffer := new(bytes.Buffer)
-	mockFileIoPerformer.On("MkdirAll", expectedService1Dir, os.FileMode(0700)).Return(nil)
-	mockFileIoPerformer.On("OpenFileReader", configFile, os.O_RDONLY, os.FileMode(0400)).Return(strings.NewReader(`{"myservice":{"edgex_use_defaults":true}}`), nil)
-	mockFileIoPerformer.On("OpenFileWriter", expectedService1File, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, os.FileMode(0600)).Return(&writeCloserBuffer{service1Buffer}, nil)
-
-	mockAuthTokenLoader := &MockAuthTokenLoader{}
-	mockAuthTokenLoader.On("Load", privilegedTokenPath).Return("fake-priv-token", nil)
-
-	expectedService1Policy := `{"path":{"secret/edgex/myservice/*":{"capabilities":["create","update","delete","list","read"]}}}`
-	expectedService1Parameters := makeDefaultTokenParameters("myservice")
-	expectedService1Parameters["meta"] = makeMetaServiceName("myservice")["meta"]
-	mockSecretStoreClient := &MockSecretStoreClient{}
-	mockSecretStoreClient.On("InstallPolicy", "fake-priv-token", "edgex-service-myservice", expectedService1Policy).Return(http.StatusNoContent, nil)
-	mockSecretStoreClient.On("CreateToken", "fake-priv-token", expectedService1Parameters, mock.Anything).
-		Run(func(args mock.Arguments) {
-			setCreateTokenResponse(args.Get(2).(*interface{}))
-		}).
-		Return(http.StatusOK, nil)
-
-	p := NewTokenProvider(mockLogger, mockFileIoPerformer, mockAuthTokenLoader, mockSecretStoreClient)
-	p.SetConfiguration(secretstoreclient.SecretServiceInfo{}, config.TokenFileProviderInfo{
-		PrivilegedTokenPath: privilegedTokenPath,
-		ConfigFile:          configFile,
-		OutputDir:           outputDir,
-		OutputFilename:      outputFilename,
-	})
-
-	// Act
-	err := p.Run()
-
-	// Assert
-	// - {OutputDir}/myservice/{OutputFilename} w/proper contents
-	// - Correct token parameters for myservice
-	// - All other expectations met
-	assert.NoError(t, err)
-	mockFileIoPerformer.AssertExpectations(t)
-	mockAuthTokenLoader.AssertExpectations(t)
-	mockSecretStoreClient.AssertExpectations(t)
-	assert.Equal(t, expectedTokenFile("myservice"), service1Buffer.Bytes())
+	_ = runTokensWithDefault("myservice", "", t)
+	_ = runTokensWithDefault("myservice", "additional-service1", t)
+	_ = runTokensWithDefault("myservice", "additional-service1,new-servcie-2", t)
+	if err := runTokensWithDefault("myservice", "additional-service1,,new-servcie-2", t); err == nil {
```

Comment:
use require.Error instead of conditional.
```
err := runTokensWithDefault("myservice", "additional-service1,,new-servcie-2", t)
require.Error(t,err, "expected error due to empty service name from the list in env")
```

Reply:
ok

Reply:
Add a test case for `,addtionalservice1,`  (leading and trailing commas)

Reply:
ok